### ISSUE TITLE
System crash on converting '.cdx' files using PyBel

### ISSUE THREAD

**Issue by theman0605** [openbabel/openbabel#1690]
- [X] I believe this to be a bug with Open Babel
- [ ] This is a feature request

## Environment Information
Open Babel version:    2.4.1
Operating system and version:  Windows 7

## Expected Behavior
Expected to be able to use PyBel from Jupyter notebook to convert ChemDraw .cdx file to .sdf or .smi file.

## Actual Behavior
System crashes and kernel stops on execution of command.  See below for script.
    

## Steps to Reproduce
 import pybel as pb
     allmols = [mol for mol in pb.readfile("cdx", "genenicStructures.cdx")

Cannot attach file due to unsupported type; see https://github.com/theman0605/cdxFileConversion for a couple of sample files for conversion.

**Comment by theman0605** [comment]
Error Messages:
![commandlineerrorimage](https://user-images.githubusercontent.com/10457951/32782316-04f77d4a-c916-11e7-8329-b5a37cd3dc3c.PNG)
![errorimage](https://user-images.githubusercontent.com/10457951/32782322-0863cf9c-c916-11e7-8e60-afc21ada1b0d.PNG)

Conversion of file using the OpenBabel GUI:
![genericstructuresscreenshot](https://user-images.githubusercontent.com/10457951/32782353-1d1f2616-c916-11e7-8b44-f24ec13365c7.PNG)

**Comment by baoilleach** [comment]
Confirmed. Reading CDX files doesn't work from the Python interface.

**Comment by ghutchis** [comment]
Yes, there seems to be a weird difference in reading binary CDX files inside another process (e.g., Python, but also Avogadro) than from the command-line. Locales maybe? Not sure.

**Comment by CamAnNguyen** [comment]
It happens on Ruby interface as well.

**Comment by serval2412** [comment]
On pc Debian x86-64 with master sources updated today, I could reproduce the pb.
Here's the bt :
[bt.txt](https://github.com/openbabel/openbabel/files/1910564/bt.txt)

**Comment by serval2412** [comment]
Here's the content of the buffer from in src/formats/chemdrawcdx.cpp:
`#6  0x00007fffe2a63a41 in OpenBabel::CDXReader::CDXReader (this=0x7fffffffdf50, is=...) at /home/julien/projects/openbabel/src/formats/chemdrawcdx.cpp:727
727	    throw;`
`(gdb) p buffer`
`$1 = "\000h!\311\342\377\177\000"`

**Comment by serval2412** [comment]
Interestingly the function is called twice and first time everything is ok.
See attached gdb session with 2 bts and 2 different values of buffer
[bt.txt](https://github.com/openbabel/openbabel/files/1910648/bt.txt)

**Comment by serval2412** [comment]
I noticed that by adding a tellg() on the input stream at the beginning of CDXReader ctr, first time I get 0, the second time I get -1
These values aren't changed even by adding a seekg(0) before calling tellg()
I wonder if the istream is still valid the second time.

**Comment by serval2412** [comment]
Keeping on the investigation, I displayed good() on the stream and as expected the second one returns 0 (so not good).
Then I displayed:
eof => 0 in both cases so ok
bad => 0 in both cases so ok
fail => 1 in the second time
This patch prevents the pb but it seems we enter an endless loop (I stopped after 6000 iterations):
`diff --git a/src/formats/chemdrawcdx.cpp b/src/formats/chemdrawcdx.cpp
index f6538cc1..e7eb591a 100644
--- a/src/formats/chemdrawcdx.cpp
+++ b/src/formats/chemdrawcdx.cpp
@@ -715,6 +715,8 @@ stringstream& CDXReader::data()
 CDXReader::CDXReader(std::istream& is) : ifs(is), depth(0)
 {
   //ReadHeader
+  ifs.clear();
+  ifs.seekg(0);
   char buffer[kCDX_HeaderStringLen+1];
   ifs.read(buffer,kCDX_HeaderStringLen);
   buffer[kCDX_HeaderStringLen] = '\0';
`

**Comment by schluta** [comment]
can confirm this is still an issue on MacOS. I can process cdx files from the command line but not from pybel.

**Comment by james-jack** [comment]
Also seems to happen with OBDotNet.  cdx => mol gives an empty molfile.
